### PR TITLE
Configuration API

### LINKED ISSUE #564: Is it Possible Extract AMQP connection string from NServiceBus.RabbitMQ
Hello,

We currently have an asp.net core 3.0 application using NServiceBus and are wanting to utilize the follow healthcheck nuget package.

https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/master/src/HealthChecks.RabbitMQ/RabbitMQHealthCheck.cs

Unfortunately, NServiceBus.RabbitMQ looks like it accepts a connection string ( which is in the format of a sql connection string ) instead of an AMQP connection string that rabbitmq expects ( https://www.rabbitmq.com/uri-spec.html ) and therefore we essentially either need to "massage" the NServiceBus RabbitMQ Connection string or have 2 connection strings in our appsettings.json

Is there any classes within NServiceBus.RabbitMQ that are part of the libraries public interface that we can use to convert the NserviceBus specific rabbitmq connection string to the mentioned AMQP connection string?

### PR DESCRIPTION
**Pull request by SeanFeldman**
Closes #564 

Adds additional methods to the configuration API that are required to be able to configure the transport when using AMQP connection string (#613).

### CODE REVIEW

**Comment by bording** [comment]
@SeanFeldman I have rebased the branch and pushed up some changes. Take a look and let me know if you have any questions.

File: src/NServiceBus.Transport.RabbitMQ/Connection/ConnectionFactory.cs [hunk 1]
```diff
@@ -48,28 +39,22 @@ class ConnectionFactory
...
+                NetworkRecoveryInterval = networkRecoveryInterval ?? connectionConfiguration.RetryDelay,
                 UseBackgroundThreadsForIO = true
             };
 
             connectionFactory.Ssl.ServerName = connectionConfiguration.Host;
-            connectionFactory.Ssl.Certs = clientCertificates;
-            connectionFactory.Ssl.CertPath = sslCertPathToUse;
-            connectionFactory.Ssl.CertPassphrase = sslCertPassPhraseToUse;
+            connectionFactory.Ssl.Certs = clientCertificateCollection;
+            connectionFactory.Ssl.CertPath = connectionConfiguration.CertPath;
```
Comment (SeanFeldman) [inline src/NServiceBus.Transport.RabbitMQ/Connection/ConnectionFactory.cs:56]: Oops, I've completely forgotten that those already exists in `connectonConfiguration`.

File: src/NServiceBus.Transport.RabbitMQ/Configuration/RabbitMQTransportSettingsExtensions.cs [hunk 2]
```diff
@@ -152,55 +152,52 @@ public static TransportExtensions<RabbitMQTransport> PrefetchCount(this Transpor
...
             Guard.AgainstNull(nameof(transportExtensions), transportExtensions);
             Guard.AgainstNullAndEmpty(nameof(path), path);
-            Guard.AgainstNullAndEmpty(nameof(passPhrase), passPhrase);
+            Guard.AgainstNullAndEmpty(nameof(password), password);
 
-            var settings = transportExtensions.GetSettings();
-
-            settings.Set(SettingsKeys.ClientCertificatePath, path);
-            settings.Set(SettingsKeys.ClientCertificatePassPhrase, passPhrase);
+            transportExtensions.GetSettings().Set(SettingsKeys.ClientCertificateCollection, new X509Certificate2Collection(new X509Certificate2(path, password)));
```
Comment (SeanFeldman) [inline src/NServiceBus.Transport.RabbitMQ/Configuration/RabbitMQTransportSettingsExtensions.cs:200]: 👍

**Comment by SeanFeldman** [comment]
Looks great, @bording. Those tweaks are good.
I can't approve since I've opened the PR, but if you think we're good, please do so.

**Pull request merged**